include/dca/phys/dca_step/cluster_solver/ctint/accumulator/ctint_accumulator.hpp

```diff
@@ -143,26 +145,38 @@ void CtintAccumulator<Parameters, device>::initialize(const int dca_iteration) {
   finalized_ = false;
 }
 
-template <class Parameters, linalg::DeviceType device>
+template <class Parameters, linalg::DeviceType device, typename Real>
 template <class Walker>
-void CtintAccumulator<Parameters, device>::updateFrom(Walker& walker) {
-  walker.computeM(M_, streams_);
+void CtintAccumulator<Parameters, device, Real>::updateFrom(Walker& walker) {
+  // Compute M.
+  walker.computeM(M_);
+
+  if (device == linalg::GPU) {
+    for (int s = 0; s < 2; ++s) {
+      event_.record(walker.get_stream(s));
+      //  Synchronize sp accumulator streams with walker.
+      event_.block(*sp_accumulator_.get_streams()[s]);
+      //  Synchronize both walker streams with tp accumulator.
+      event_.block(*tp_accumulator_.get_stream());
+    }
+  }
+
   configuration_ = walker.getConfiguration();
-  sign_ = walker.getSign();
+  sign_ = walker.get_sign();
```

Comment:
Why revert to legacy naming for function?

Reply:
I had to bring the ctint walker interface for the sign in line with the other solvers. Or I could have modified the other two solvers. get/set_<variable name> is accepted in the code style for trivial getter setters, and it seems to be the preferred  way. I admit that my code and  the pre-existing is not very consistent on this and we should agree on a convention to follow for the future

Reply:
This part of our coding style is problematic in that it leaks knowledge of the class implementation, in this case it is a simple getter